src/coreclr/vm/perfmap.cpp

```diff
@@ -49,7 +57,9 @@ void PerfMap::Initialize()
         }
 
         s_enabled = true;
```

Comment:
I was under the impression that the methods in perfmap.cpp were only used/needed for the generation of the perfmap file. The generation of the jit dump file has its own methods in the perfjitdump file, for example `LogMethod`:
https://github.com/dotnet/runtime/blob/53bb518ea482a5960577b151c11cc1297e5dea6d/src/coreclr/pal/src/misc/perfjitdump.cpp#L233-L321

This would mean that s_enabled would only be needed for when perfmap is enabled, or am I missing something?

Reply:
The implementation of jitdump is in perfjitdump.cpp, but you have to call `PAL_PerfJitDump_LogMethod` to actually write to the jitdump file, and the call to `PAL_PerfJitDump_LogMethod` only happens in `PerfMap::LogMethod` and it is behind a check for `s_enabled` through `PerfMap::LogJITCompiledMethod` and `PerfMap::LogPreCompiledMethod`

I haven't run your changes, but I would expect you would get an empty jitdump file as it stands

Reply:
I see, I didn't realize the call to `PAL_PerfJitDump_LogMethod` is in `PerfMap::LogPreCompiledMethod`. A fix to make it work should be simple, but it seems less than ideal to keep the calls to write to the JitDump file in the call to the write to the PerfMap

Reply:
@davmason this is only really a problem in `PerfMap::LogJITCompiledMethod` since the call to `PAL_PerfJitDump_LogMethod` is behind a call to `s_Current->LogMethod`. Can you think of an easier way to fix this other than cutting the code in `LogMethod()` and putting it into `LogJITCompiledMethod`?

Reply:
I think getting rid of `LogMethod` and folding it in to `LogJITCompiledMethod` is fine

Reply:
These changes have been made.